homeassistant/components/tesla_fleet/__init__.py

```diff
@@ -64,14 +64,22 @@
 async def async_setup_entry(hass: HomeAssistant, entry: TeslaFleetConfigEntry) -> bool:
     """Set up TeslaFleet config."""
 
+    try:
+        implementation = await async_get_config_entry_implementation(hass, entry)
+    except (ValueError, KeyError) as e:
```

Comment:
I believe only ValueError is thrown here with "Implementation not available". Can `KeyError` be omitted?

Reply:
Your right, KeyError was only thrown if the config entry has no auth_implementation at all. I changed data to `{"auth_implementation": None}` to avoid triggering KeyError. I can drop it.